### DISCUSSION TITLE
Providing links to publishDir outputs

### DISCUSSION THREAD

**Discussion by vpresnyakLM** [nextflow-io/nextflow#2426] [Q&A]
Hello all,

I'm looking to understand if there is a way to get paths to output files created by `publishDir`. I have a pipeline doing a number of operations on files and publishing some intermediates. I am looking to create a table with links to the published files to insert into our tracking database for easy access when doing meta analyses.

My process looks something like the code below. What would be the best way to get the full published file paths?

```
process STAR_ALIGN {
    
    publishDir "${params.outdir}/${params.project_id}/", mode: "copy", pattern: "${sample}.*bam"
    
    input:
    val(sample)
    path(star_ref_dir)
    tuple path(fq_r1), path(fq_r2)

    output:
    val(sample),                                     emit: sample
    path(star_ref_dir),                              emit: ref_dir
    path("${sample}.Aligned.sortedByCoord.out.bam"), emit: bam
    path("${sample}.Log.final.out"),                 emit: final_log

    script:
    """
    STAR ...
    """
}
```

**Comment by bentsherman** [comment]
Hi @vpresnyakLM , answering for posterity here even though it's been a while. Since publishing is handled by the head job separately from the tasks, the only way to do what you want currently is to output the publish paths to a channel and build the table in a downstream process.
